src/PrestaShopBundle/Form/Admin/Sell/Attachment/AttachmentType.php

```diff
@@ -0,0 +1,135 @@
+use PrestaShopBundle\Translation\TranslatorAwareTrait;
+use Symfony\Component\Form\AbstractType;
+use Symfony\Component\Form\Extension\Core\Type\FileType;
+use Symfony\Component\Form\Extension\Core\Type\TextType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\Form\FormInterface;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+use Symfony\Component\Validator\Constraints\Length;
+use Symfony\Component\Validator\Constraints\NotBlank;
+
+/**
+ * Attachment form type definition
+ */
+class AttachmentType extends AbstractType
+{
+    use TranslatorAwareTrait;
+
+    /**
+     * {@inheritdoc}
+     */
+    public function buildForm(FormBuilderInterface $builder, array $options)
+    {
+        $builder
+            ->add('name', TranslatableType::class, [
+                'type' => TextType::class,
+                'required' => true,
+                'options' => [
+                    'constraints' => [
+                        new TypedRegex(
+                            [
+                                'type' => 'generic_name',
+                            ]
+                        ),
+                    ],
+                    new Length(
+                        [
+                            'max' => AttachmentValidationConfiguration::MAX_NAME_LENGTH,
+                            'maxMessage' => $this->trans(
+                                'This field cannot be longer than %limit% characters',
+                                ['%limit%' => AttachmentValidationConfiguration::MAX_NAME_LENGTH],
+                                'Admin.Notifications.Error'
+                            ),
+                        ]
+                    ),
+                ],
+                'constraints' => [
+                    new DefaultLanguage(),
+                ],
+            ])
+            ->add('file_description', TranslatableType::class, [
+                'type' => TextType::class,
+                'required' => false,
+                'options' => [
+                    'constraints' => [
+                        new CleanHtml(),
+                    ],
+                ],
+            ])
+            ->add('file', FileType::class, [
+                'required' => !$options['is_edit_form'] ||
+                    (isset($options['has_old_file']) && !$options['has_old_file']),
+                'constraints' => [
+                    new NotBlank([
+                        'message' => $this->trans('This field cannot be empty', [], 'Admin.Notifications.Error'),
+                        'groups' => ['validate'],
+                    ]),
+                ],
+            ])
+        ;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function configureOptions(OptionsResolver $resolver)
+    {
+        $resolver->setRequired(['is_edit_form', 'has_old_file']);
+        $resolver->setDefaults([
+            'validation_groups' => function (FormInterface $form) {
```

Comment:
this seems very complex 🤔 can you explain what is the reason behind all this logic ?

Using a closure here seems a bit too much 😅

Reply:
I need to determine if validation is needed based on submitted data and this is what symfony suggests doing for dynamic validation groups. https://symfony.com/doc/current/form/data_based_validation.html. I can't think of a simpler solution maybe you have something in mind ? :)

Reply:
All this triggers whether or not the `NotBlank` validation on FileType input is enabled ?

If it's this, then I feel like it's overkill 😅: in order to check whether given file is not blank, we first check which validation group should be used which is found out thanks to a closure who checks the data ...

Maybe we can just perform the `NotBlank` check at a later stage of the process, like in the handler ? `If file is blank, throw EmptyFileException`.

If the validation used was something trickier thant a `NotBlank`, like a check against non-conventional chars, file size, file type, and we *expect* the user to logically miss this so we want to give him a nice form error, alright we could do all that. But if someone submits a blank file ... he is really dumb 😅I dont expect standard users to do this. That's why I think it might be a little too much to make sure this dumb user is fine.